manual_process_work/vbuild/Vtop__Slow.cpp

```diff
@@ -0,0 +1,376 @@
+
+    p_second.run_conditions.push_back([](Vtop__Syms* __restrict vlSymsp) {
+                return vlSymsp->ready_to_finish.triggered;
+            });
+}
+
+void s_second_s_3(Vtop__Syms* __restrict vlSymsp) {
+    printf("After ready to finish: i=%d, j=%d, job=%p\n",
+           vlSymsp->i, vlSymsp->j, vlSymsp->job);
+
+    VL_WRITEF("All done!\n");
+    VL_FINISH_MT("top.sv", 44, "");
+}
+
+void s_initial_0_s_0(Vtop__Syms* __restrict vlSymsp) {
+    // #500;
+
+    p_initial_0.time_conditions.push_back(timer.get_slot(500));
+}
+
+void s_initial_0_s_1(Vtop__Syms* __restrict vlSymsp) {
+    // $display("Should not reach this before $finish");
+    // $stop;
+
+    VL_WRITEF("Should not reach this before $finish\n");
+    VL_STOP_MT("top.sv", 51, "");
+}
+
+//==========
+
+VL_CTOR_IMP(Vtop) {
+    Vtop__Syms* __restrict vlSymsp = __VlSymsp = new Vtop__Syms(this, name());
+    Vtop* const __restrict vlTOPp VL_ATTR_UNUSED = vlSymsp->TOPp;
+    // Reset internal values
+    
+    // Reset structure values
+    _ctor_var_reset();
+}
+
+void Vtop::__Vconfigure(Vtop__Syms* vlSymsp, bool first) {
+    if (false && first) {}  // Prevent unused
+    this->__VlSymsp = vlSymsp;
+    if (false && this->__VlSymsp) {}  // Prevent unused
+    Verilated::timeunit(-12);
+    Verilated::timeprecision(-12);
+}
+
+Vtop::~Vtop() {
+    VL_DO_CLEAR(delete __VlSymsp, __VlSymsp = NULL);
+}
+
+void Vtop::_initial__TOP__1(Vtop__Syms* __restrict vlSymsp) {
+    VL_DEBUG_IF(VL_DBG_MSGF("+    Vtop::_initial__TOP__1\n"); );
+    Vtop* const __restrict vlTOPp VL_ATTR_UNUSED = vlSymsp->TOPp;
+    // Body
+
+    // FIXME initializing ExecutionBlocks should be done outside of 'eval'
+    // in some cases (all?) even dynamically (especially when forking to
+    // pre-defined tasks etc)
+
+    p_initial.add_step(s_initial_s_0);
+
+    {
+        p_first.add_step(s_first_s_0);
+        p_first.add_step(s_first_s_1);
+
+        p_second.add_step(s_second_s_0);
+        {
+            p_second_0.add_step(s_second_0_s_0);
+
+            p_second_1.add_step(s_second_1_s_0);
+            p_second_1.add_step(s_second_1_s_1);
+            p_second_2.add_step(s_second_2_s_0);
+            {
+                p_second_2_0.add_step(s_second_2_0_s_0);
+                p_second_2_0.add_step(s_second_2_0_s_1);
+            }
+            p_second_2.add_step(s_second_2_s_1);
+        }
+        p_second.add_step(s_second_s_1);
+        p_second.add_step(s_second_s_2);
+        p_second.add_step(s_second_s_3);
+
+        p_initial_0.add_step(s_initial_0_s_0);
+        p_initial_0.add_step(s_initial_0_s_1);
+    }
+
+    // starting only the initial block (it should start the other blocks
+    // itself)
+
+    active_processes.insert(&p_initial);
+
+    while (active_processes.size() > 0) {
+        unsigned int run_count = 0;
+
+        for (auto p: active_processes) {
+            bool can_run = true;
+            for (auto c: p->run_conditions) {
+                can_run &= c(vlSymsp);
```

Comment:
Suspect searching a list every cycle will be too inefficient once the list of processes gets large, we'll need to think about a better way to represent it.